v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/edr-provider.ts

```diff
@@ -80,9 +83,16 @@ export const DEFAULT_EDR_NETWORK_HD_ACCOUNTS_CONFIG_PARAMS: EdrNetworkDefaultHDA
   };
 
 export async function isDefaultEdrNetworkHDAccountsConfig(
-  account: unknown,
+  accounts: EdrNetworkHDAccountsConfig,
 ): Promise<boolean> {
-  return deepEqual(account, DEFAULT_EDR_NETWORK_HD_ACCOUNTS_CONFIG_PARAMS);
+  return deepEqual(
+    {
+      ...accounts,
+      mnemonic: await accounts.mnemonic.get(),
+      passphrase: await accounts.passphrase.get(),
+    },
+    DEFAULT_EDR_NETWORK_HD_ACCOUNTS_CONFIG_PARAMS,
+  );
 }
```

Comment:
I had to update this function as before, it would always return false since the types of `menominic` and `passphrase` were different in `EdrNetworkHDAccountsConfig` and `DEFAULT_EDR_NETWORK_HD_ACCOUNTS_CONFIG_PARAMS`.

Reply:
Uh, I've addressed this differently in a PR that was merged today. Instead of delegating the resolution of the config variables to the compare function, I used the getters before calling the function. You can check it out here: [link](https://github.com/NomicFoundation/hardhat/blob/a3e1f123be84a06e0406631a7a36c6c05909323b/v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/utils/convert-to-edr.ts#L258). This is temporary though, as EDR network accounts may not use config variables in the future. See https://github.com/NomicFoundation/hardhat/issues/6210

Reply:
I see! Yeah, that works too. Would you prefer for the resolution to stay outside `isDefaultEdrNetworkHDAccountsConfig`? Personally, I like the fact that by moving it to the function, we can type the input argument.

Reply:
Either way it's fine. If you're going to keep this implementation, could you rebase and update the other call?

Reply:
Of course! And thank you for the review 🙇 I'll add a test for this as well.